### PR TITLE
feat: Support for tier 1 teams

### PR DESCRIPTION
**Pull request by AndyLnd**

### CODE REVIEW

File: src/script/user/UserRepository.ts [hunk 1]
```diff
@@ -219,22 +221,13 @@ export class UserRepository {
 
   async loadUsers(): Promise<void> {
     if (this.isTeam()) {
-      if (this.isTeamTooLargeForBroadcast()) {
-        this.logger.warn(
-          `Availability not displayed since the team size is larger or equal to "${UserRepository.CONFIG.MAXIMUM_TEAM_SIZE_BROADCAST}".`,
-        );
-        return;
-      }
-
```
Comment (AndyLnd) [inline src/script/user/UserRepository.ts:230]: This guard is not needed anymore, as we don't limit the status update in the same way as we used to.

File: src/script/user/UserRepository.ts [hunk 2]
```diff
@@ -281,27 +274,20 @@ export class UserRepository {
...
         data: {availability},
       } = event;
-
-      if (userId !== this.self().id && this.isTeamTooLargeForBroadcast()) {
-        this.logger.warn(
-          `Availability not updated since the team size is larger or equal to "${UserRepository.CONFIG.MAXIMUM_TEAM_SIZE_BROADCAST}".`,
-        );
-      } else {
-        this.get_user_by_id(userId).then(userEntity => userEntity.availability(availability));
-      }
```
Comment (AndyLnd) [inline src/script/user/UserRepository.ts:293]: The same as before. We handle every availability update that we receive.

File: src/script/user/UserRepository.ts [hunk 3]
```diff
@@ -347,23 +333,18 @@ export class UserRepository {
     });
   }
 
-  private isTeamTooLargeForBroadcast(): boolean {
-    const teamSizeIncludingSelf = this.teamUsers().length + 1;
-    return teamSizeIncludingSelf >= UserRepository.CONFIG.MAXIMUM_TEAM_SIZE_BROADCAST;
-  }
-
```
Comment (AndyLnd) [inline src/script/user/UserRepository.ts:356]: Not necessary anymore. This was only used for limiting the availability updates.

File: src/script/user/UserRepository.ts [hunk 4]
```diff
@@ -425,14 +404,13 @@ export class UserRepository {
       messageId: createRandomUuid(),
     });
 
-    const recipients = this.isTeamTooLargeForBroadcast() ? [this.self()] : teamUsers.concat(this.self());
-    amplify.publish(WebAppEvents.BROADCAST.SEND_MESSAGE, {genericMessage, recipients});
+    const sortedUsers = this.directlyConnectedUsers().sort(({id: idA}, {id: idB}) =>
+      idA.localeCompare(idB, undefined, {sensitivity: 'base'}),
+    );
+    const [members, other] = partition(sortedUsers, user => user.isTeamMember());
+    const recipients = [this.self(), ...members, ...other].slice(0, UserRepository.CONFIG.MAXIMUM_TEAM_SIZE_BROADCAST);
```
Comment (AndyLnd) [inline src/script/user/UserRepository.ts:418]: The recipients of the status update are now determined as: [self, ...team members, ...other connected users]. This list is  cut after 500. To make sure that every platform would send to the same 500 users, they are sorted by their user id.

File: src/script/components/userList.ts [hunk 5]
```diff
@@ -174,29 +177,44 @@ ko.components.register('user-list', {
       }
     };
 
+    const remoteTeamMembers = ko.observable([]);
+
+    async function fetchMembers(query: string, isHandle: boolean, ignoreMembers: User[]) {
```
Comment (bennycode) [inline src/script/components/userList.ts]: Let's comment what is the purpose of this function.

File: src/script/conversation/ClientMismatchHandler.ts [hunk 6]
```diff
@@ -176,40 +185,36 @@ export class ClientMismatchHandler {
...
 
-      if (noRemainingClients) {
-        if (conversationEntity !== undefined) {
-          if (conversationEntity.isGroup() && verifyDeletedUsers) {
-            const timestamp = this.serverTimeHandler.toServerTimestamp();
-            const event = EventBuilder.buildMemberLeave(conversationEntity, userId, false, timestamp);
-            this.eventRepository.injectEvent(event);
-          }
+      if (noRemainingClients && typeof conversationEntity !== 'undefined' && conversationEntity.isGroup()) {
+        const result = await this.userRepository['userService'].getUser(userId);
```
Comment (bennycode) [inline src/script/conversation/ClientMismatchHandler.ts]: Dangerous as `userService` seems to be private? Let's expose the functionality through the `userRepository`.

File: src/script/view_model/list/StartUIViewModel.js [hunk 7]
```diff
@@ -377,6 +368,34 @@ class StartUIViewModel {
...
 
       this.searchResults.groups(this.conversationRepository.getGroupsByName(normalizedQuery, isHandle));
+
+      if (!this.showOnlyConnectedUsers()) {
+        try {
+          const userEntities = await this.searchRepository.search_by_name(normalizedQuery, isHandle);
+
+          const isCurrentQuery = normalizedQuery === SearchRepository.normalizeQuery(this.searchInput());
+          if (isCurrentQuery) {
+            if (this.selfUser().inTeam()) {
```
Comment (bennycode) [inline src/script/view_model/list/StartUIViewModel.js]: Huge block of business logic. Let's move it into a separate function so it can be easily tested.

**Pull request merged**